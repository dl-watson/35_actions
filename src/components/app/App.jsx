import React from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { Comments } from "../comments/Comments";
import { Posts } from "../posts/Posts";

const App = () => {
  return (
    <Router>
      <Switch>
        <Route exact path="/" component={Posts} />
        <Route path="/:index" component={Comments} />
      </Switch>
    </Router>
  );
};

export default App;
