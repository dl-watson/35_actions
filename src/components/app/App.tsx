import React from "react";
import { BrowserRouter as Router, Route, RouteComponentProps, Switch } from "react-router-dom";
import { Comments } from "../comments/Comments";
import { Posts } from "../posts/Posts";

const App = () => {
  return (
    <Router>
      <Switch>
        <Route exact path="/" component={Posts} />
        <Route path="/:index" component={Comments} />
      </Switch>
    </Router>
  );
};


interface MatchParams {
  index?: string;
}

interface Props extends RouteComponentProps<MatchParams> {
}

export default App;
