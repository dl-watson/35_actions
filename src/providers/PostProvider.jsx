import React, { createContext, useReducer } from "react";
import { initialState, postReducer } from "../reducers/postReducer";

const PostContext = createContext(null);

export const PostProvider = ({ children }) => {
  const [state, dispatch] = useReducer(postReducer, initialState);

  return <PostContext.Provider>{children}</PostContext.Provider>;
};
