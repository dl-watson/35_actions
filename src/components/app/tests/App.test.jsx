import React from "react";
import { render, cleanup } from "@testing-library/react";
import App from "../App";
import { PostProvider } from "../../../providers/PostProvider";

describe("App component", () => {
  afterEach(() => cleanup());
  it("renders App", () => {
    const { asFragment } = render(
      <PostProvider>
        <App />
      </PostProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
