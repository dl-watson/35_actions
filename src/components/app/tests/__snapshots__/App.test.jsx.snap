// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <section
    class="NewPostForm"
  >
    <h2>
      Create Post
    </h2>
    <form>
      <input
        name="title"
        placeholder="Title..."
        value=""
      />
      <textarea
        name="body"
        placeholder="Body..."
      />
      <button>
        Submit
      </button>
    </form>
  </section>
  <ul />
</DocumentFragment>
`;
